apps/omg_performance/lib/omg_performance/byzantine_events.ex

```diff
@@ -163,8 +137,32 @@ defmodule OMG.Performance.ByzantineEvents do
     _ = Logger.info("Watcher synchronized")
   end
 
+  def get_byzantine_events(event_name) do
+    watcher_url = Application.fetch_env!(:omg_performance, :watcher_url)
+    {:ok, status_response} = WatcherClient.get_status(watcher_url)
+
+    status_response
+    |> Access.get(:byzantine_events)
+    |> Enum.filter(&(&1["event"] == event_name))
+    |> postprocess_byzantine_events(event_name)
+  end
+
+  defp postprocess_byzantine_events(events, "invalid_exit"), do: Enum.map(events, & &1["details"]["utxo_pos"])
+
   defp only_successes(responses), do: Enum.map(responses, fn {:ok, response} -> response end)
 
+  # this allows one to map a contract-transacting function over a collection nicely.
+  # It initiates all the transactions concurrently. Then it waits on all of them to mine successfully.
+  # Returns the last receipt result, so you can synchronize on the block number returned (and the entire bundle of txs)
+  defp map_contract_transaction(enumberable, transaction_function) do
+    enumberable
+    |> Enum.map(transaction_function)
+    # NOTE: infinity doesn't work, hence the large number
+    |> Task.async_stream(&Support.DevHelper.transact_sync!/1, max_concurrency: 10_000)
+    |> Enum.map(fn {:ok, result} -> result end)
+    |> List.last()
```

Comment:
It isn't obvious to me that the particular last transaction in the list completes the last.
Is it possible that but-last tx completes as last one - but we'll be waiting "little" shorter (?)

Reply:
ha, I stopped at this too. But they will actually all complete (`Task.async_stream` isn't that async :) ).

What's more, it will also return the last transaction sent out - note that the actual sending of transactions is synchronous and sequentioal (`Enum.map(transaction_function)`). Only the "waiting" and "receipt getting" is `Task.async_stream`, and there, order is the same as with inputs :sunglasses: